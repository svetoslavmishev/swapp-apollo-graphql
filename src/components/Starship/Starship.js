import React from 'react';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/react-hooks';
import { makeStyles } from '@material-ui/core/styles';
import {
  Container,
  Card,
  CardHeader,
  CardActionArea,
  CardMedia,
  CardContent,
  Typography
} from '@material-ui/core';
import { GET_STARSHIP } from '../../queries/queries';

const useStyles = makeStyles(theme => ({
  card: {
    maxWidth: 300,
    margin: 30
  },
  media: {
    height: 0,
    paddingTop: '100%'
  }
}));

const Starship = () => {
  const { starshipId } = useParams();
  const { data, loading, error } = useQuery(GET_STARSHIP, {
    variables: { id: starshipId }
  });
  const classes = useStyles();

  if (loading) return <div>Loading...</div>;
  if (error) return null;
  const {
    starship: {
      id,
      name,
      model,
      image,
      starshipClass,
      cost,
      crew,
      maxAtmosphericSpeed,
      hyperdriveRating
    }
  } = data;

  return (
    <>
      <Container className={classes.root} maxWidth="xl">
        <div className="flex justify-center bg-green-400">
          <div>{name}</div>
          <div>{model}</div>
        </div>
        <Card className={classes.card} id={id}>
          <CardActionArea>
            <CardHeader title={name} />
            <CardMedia className={classes.media} image={image} title={name} />
            <CardContent>
              <Typography variant="body2" component="p">
                Class: {starshipClass}
              </Typography>
              <Typography variant="body2" component="p">
                Cost: {cost} credits
              </Typography>
              <Typography variant="body2" component="p">
                Crew: {crew}
              </Typography>
              <Typography variant="body2" component="p">
                Max Atmospheric Speed: {maxAtmosphericSpeed}
              </Typography>
              <Typography variant="body2" component="p">
                Hyperdrive Rating: {hyperdriveRating}
              </Typography>
            </CardContent>
          </CardActionArea>
        </Card>
      </Container>
    </>
  );
};

export default Starship;
