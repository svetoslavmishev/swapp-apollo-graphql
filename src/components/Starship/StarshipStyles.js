import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    flexWrap: 'wrap',
    minHeight: 'calc(100vh - 64px)',
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].backgroundColor
  },
  card: {
    maxWidth: 300,
    margin: 30
  },
  media: {
    height: 0,
    paddingTop: '100%'
  },
  title: {
    fontFamily: 'StarWars',
    color: ({ currentTheme }) =>
      theme.palette[currentTheme].PrimaryHeading.fontColor
  },
  subTitle: {
    fontFamily: 'StarWars',
    color: theme.palette.commonColors.lightBlue,
    fontSize: 16
  }
}));

export default styles;
