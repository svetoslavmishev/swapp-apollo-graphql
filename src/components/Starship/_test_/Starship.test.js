import React from 'react';
import { shallow } from 'enzyme';
import { ThemeContext, theme } from '../../../themeContext';
import Starship from '../Starship';

describe('<Starship />', () => {
  it('renders <Starship /> without error', () => {
    const wrapper = shallow(
      <ThemeContext.Provider value={theme}>
        <Starship />
      </ThemeContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
