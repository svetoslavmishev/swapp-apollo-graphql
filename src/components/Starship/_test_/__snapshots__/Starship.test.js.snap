// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Starship /> renders <Starship /> without error 1`] = `ShallowWrapper {}`;
